### PR TITLE
[Windows] support Npcap

### PR DESCRIPTION
**Pull request by gpotter2**
Winpcap hasn't been updated since 2013. This PR add support for Npcap and display a warning encouraging people to migrate to Npcap.

This also means that we have to use the updated windump version.

Npcap is a relevant program, part of the nmap project, which brings functions like loopback adapter and update the code for newer versions. The windump version is made by the author of Npcap.

Also has several fixes:
- Loopback Routing fixes (as Npcap supports loopback)
- Updated` route.py` \_\_repr\_\_ with the one located in `route6.py` (better spaces management)
- Fixed mock_windows.uts with a way easier tests

### CODE REVIEW

**Comment by hsluoyz** [comment]
@gpotter2 , the bug has been fixed in latest WinDump 0.2, please try it at: https://github.com/hsluoyz/WinDump/releases

**Comment by hsluoyz** [comment]
Another issue I found that the current scapy doesn't support the Npcap native mode (Npcap DLLs will be installed in ``System32\Npcap`` instead of ``System32``).  The impact is that the user has to install Npcap with ``WinPcap Compatible Mode`` option checked to use it with scapy. Otherwise, scapy cannot find Npcap DLLs. And enabling this option will install another copy of Npcap driver that is compatible with WinPcap (like installing the DLLs to ``System32``), which may affect the performance. **So Npcap suggests all its downstream software to support Npcap native mode.** Now software like Wireshark, Nmap, GNS3 has already supported Npcap native mode. Please refer to the docs [here](https://rawgit.com/nmap/npcap/master/docs/npcap-guide-wrapper.html#npcap-feature-native) for details and implementation guidelines.

File: scapy/arch/winpcapy.py [hunk 1]
```diff
@@ -7,31 +8,34 @@
...
 HAVE_REMOTE=False
 
-
-if sys.platform.startswith('win'):
-    WIN32=True
+if WINDOWS:
     HAVE_REMOTE=True
-
-if WIN32:
     SOCKET = c_uint
```
Comment (gpotter2) [inline scapy/arch/winpcapy.py]: @hsluoyz Don't know if you noticed, but that was modified :)

File: scapy/arch/winpcapy.py [hunk 2]
```diff
@@ -7,31 +8,34 @@
...
-    _lib=CDLL('wpcap.dll')
+    npcap_folder = os.environ["WINDIR"] + "\\System32\\Npcap"
+    if os.path.exists(npcap_folder):
+        # Try to load npcap
+        os.environ['PATH'] = npcap_folder + ";" + os.environ['PATH']
+    _lib=CDLL("wpcap.dll")
```
Comment (gpotter2) [inline scapy/arch/winpcapy.py]: @hsluoyz Don't know if you noticed, but that was modified :)

**Comment by gpotter2** [comment]
@p-l- Please add https://github.com/secdev/scapy/pull/526 before

**Comment by guedou** [comment]
@gpotter2 is this patch needed ? I thought that npcap was compatible with wpcap. I am already using it with Scapy and it works fine.

**Comment by gpotter2** [comment]
@guedou Well as @hsluoyz said, native Npcap support requires some fixes as `winpcap.dll` is not located in the same path.
Then this patch fixes a few bugs, recommand the users to switch from winpcap to npcap and finally migrates appveyor from winpcap to npcap.

So I would say so

**Comment by gpotter2** [comment]
@p-l- hello,
 it would never great if this one could be the next one to be reviewed...
Thanks !

**Comment by p-l-** [comment]
You have conflicts. Please `git rebase -i` against current master so that I can review the exact changes.

**Comment by gpotter2** [comment]
Conflicts resolved.

File: scapy/arch/windows/__init__.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 """
 Customizations needed to support Microsoft Windows.
 """
-import os,re,sys,socket,time, itertools
+import os, re, sys, socket, time, itertools, platform, subprocess
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: Warning: you are importing subprocess twice here!

File: scapy/arch/windows/__init__.py [hunk 4]
```diff
@@ -220,7 +233,6 @@ def win_find_exe(filename, installsubdir=None, env="ProgramFiles"):
             break        
     return path
 
-
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: Please avoid unnecessary changes.

**Comment by gpotter2** [comment]
@p-l- Ready to merge (may want to restart the single failing travis test)
